### DISCUSSION TITLE
Builds & GitHub's free tier limits

### DISCUSSION THREAD

**Discussion by v1gnesh** [zopencommunity/meta#540] [General]
Don't know in which repo, but one of them failed after 6h; that appears to be the job execution time limit.
https://docs.github.com/en/actions/learn-github-actions/usage-limits-billing-and-administration#usage-limits

In addition, we have 33.3 hours/month in Actions run-time in the free tier.
https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions#included-storage-and-minutes

Instead of keeping the build & test action active, maybe it just kicks off the job, pauses the PR or whatever, and ends the Action?
Then with another daily Action with a timer, we can update open PRs with the statuses from CI.

**Comment by IgorTodorovskiIBM** [comment]
Thanks, that's too bad.

Yeah, I'm thinking I can split the github build and test action into two parts, one that spawns the jenkins build & test job (should only run for a few seconds) and then another which is called from the Jenkins job upon completion which would update the status check and attach the console log.

**Comment by IgorTodorovskiIBM** [comment]
I believe the 2000minutes/month only applies to private repositories:
> GitHub Actions usage is free for standard GitHub-hosted runners in public repositories, and for self-hosted runners. For **private repositories**, each GitHub account receives a certain amount of free minutes and storage for use with GitHub-hosted runners, depending on the account's plan. Any usage beyond the included amounts is controlled by spending limits.

The 6 hour execution limit is probably an issue and will affect autoconf (takes around 11 hours) and cmake (takes around 12 hours). Once we start using real zHW, then it may no longer be an issue.

**Comment by MikeFultonDev** [comment]
Closing this off because we don't have any limits because we only have public repos
